lib/utils.js

```diff
@@ -1,37 +1,61 @@
-const { curry } = require('ramda');
-const { KEY_START, KEY_END, RSA_KEY_START, RSA_KEY_END } = require('./constants');
-const domainRegTest = /^[a-z0-9-]+$/;
-
-const tryParseJson = (keyjson) => {
-  if (typeof keyjson === 'string') {
-    try {
-      keyjson = JSON.parse(keyjson);
-    } catch (e) {
-      throw Error(`Could not parse json properly: ${e.message}`);
-    }
-  }
-  return keyjson;
-};
+const {
+  curry, allPass, anyPass, map, startsWith, endsWith, find, trim
+} = require('ramda');
+const { KEY_TYPES, CERT_TYPES } = require('./constants');
 
-const isSslKey = (key) => {
-  return !(key.startsWith(KEY_START) && key.endsWith(KEY_END)) && !(key.startsWith(RSA_KEY_START) && key.endsWith(RSA_KEY_END))
-};
+const KEY_PAIRS = KEY_TYPES.map((type) => {
+  return { start: `-----BEGIN ${type}-----`, end: `-----END ${type}-----` };
+});
+
+const CERTIFICATE_PAIRS = CERT_TYPES.map((type) => {
+  return { start: `-----BEGIN ${type}-----`, end: `-----END ${type}-----` };
+});
+
+const buildValidFormatFunc = curry(map(({ start, end }) => {
```

Comment:
What's `curry` doing for you here? `map` is already curried and should be waiting for an arg. Maybe I'm missing something.